pkgs/applications/misc/watershot/default.nix

```
@@ -0,0 +1,31 @@
+{ lib, rustPlatform, fetchFromGitHub, pkg-config, wayland, libxkbcommon
+, fontconfig, makeWrapper, grim }:
+rustPlatform.buildRustPackage rec {
+  pname = "watershot";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "Kirottu";
+    repo = "watershot";
+    rev = "v${version}";
+    hash = "sha256-8GqO7Y0d+AoYr3Us3FEfNobrQNSw7XyGwmZz5HkVvDg=";
+  };
+
+  cargoLock.lockFile = src + "/Cargo.lock";
```

That's an IFD (import from derivation), which isn't allowed in nixpkgs.

In rare cases where `cargoLock.lockFile` is needed, because `cargoHash` does not work, the `Cargo.lock` needs to be upstreamed into nixpkgs.

But this isn't the case here :)

```suggestion
  cargoHash = "sha256-yJD7c/I3rwzczcrxbD8sinzP7bjMzhWWAVcCFCsTdeo=";
```